runtime/gc_vlhgc/AllocationContextBalanced.hpp

```diff
@@ -72,6 +72,7 @@ class MM_AllocationContextBalanced : public MM_AllocationContextTarok
 	UDATA _freeProcessorNodeCount;	/**< The length, in elements, of the _freeProcessorNodes array (always at least 1 after startup) */
 
 #if defined(J9VM_GC_SPARSE_HEAP_ALLOCATION)
+	intptr_t _fractionReservedRegion;
```

Comment:
I'd suggest renaming this to 
_currentSharedArrayReservedRegionBytesUsed

I'd also introduce another variable:
_sharedArrayReserveRegionsCount that would be properly incremented/decremented  needAllocateReservedRegionFraction/needRecycleReservedRegionFraction when we move to another shared region
That variable would be useful for debugging, statistics, and would possibly simplify some other code (like ProjectedSurvival math).

Reply:
The current way when we are operating with fractional part of the region looks easier to understand and have deal with in general. But it means maintaining two counters (one for stats) instead of one (and these counters can be off potentially). We can do either way.